orttraining/orttraining/training_ops/cuda/nn/layer_norm_impl.cu

```diff
@@ -409,8 +422,8 @@ __device__ void cuLoadWriteStridedInputs(
           warp_buf2[write_idx] = curr_dout * (curr_input - curr_mean) * curr_invvar;
         } else {
           U curr_gamma = static_cast<U>(gamma[i2]);
-          U curr_beta = static_cast<U>(beta[i2]);
-          U curr_output = static_cast<U>(output[load_idx]);
+          U curr_beta = beta != nullptr ? static_cast<U>(beta[i2]) : U(0);
+          U curr_output = output != nullptr ? static_cast<U>(output[load_idx]) : U(0);
```

Comment:
if output is a nullptr, will the calculation below ever be correct? This code path is reached when we want to use the output instead of the input in grad calculation and we must not continue computation when output is not available.

Reply:
This is the invertible LM path. I've reverted changes on this path